Essence of Linear Algebra (#357)
3b1b/3Blue1Brown.com: public/content/lessons/2016/cross-products-extended/index.mdx

---
title: Cross products in the light of linear transformations
description: The formula for the cross product can feel like a mystery, or some kind of crazy coincidence. But it isn't. There is a fundamental connection between the cross product and determinants.
date: 2016-09-01
chapter: 11
video: BaM7OCEm3G0
source: _2016/eola/chapter8.py
credits:
- Lesson by Grant Sanderson
---

> "From [Grothendieck], I have also learned not to take glory in the difficulty of a proof: difficulty means we have not understood. The idea is to be able to paint a landscape in which the proof is obvious."
>
> \- Pierre Deligne

> Where we left off, I was talking about how to compute a three-dimensional cross product of two vectors, v cross w. It's this funny thing where you write a matrix whose second column has the coordinates of v, whose third columns has the coordinates of w, but the entries of the first column, weirdly, are the basis vectors i hat, j hat and k hat, where you just pretend like those guys are numbers for the sake of computations. If you just chug along with the computation, ignoring this weirdness, you get some constant times i hat, plus some constant times j hat, plus some constant times k hat, which defines a new 3d vector.

Reviewer: Replace some of the prose description of vectors with tex, 
- "v cross w" -> "$\mathbf{v} \times \mathbf{w}$"
- "i hat" -> "$\hat{imath}$"
- etc.

<Figure
    image="figures/introduction-determinant-with-basis-vectors.png" 
/>

From here, students are typically told to just believe that the resulting vector has the following geometric properties: 

- Its length equals the area of the parallelogram defined by v and w. 
- It points in a direction perpendicular to v and w,

> - And this direction obeys the right-hand rule, in the sense that if you point your forefinger along v, and your middle finger along w, then stick out your thumb, it will point in the direction of the new vectors.

Reviewer: ```suggestion
- This direction obeys the right-hand rule, in the sense that if you point your forefinger along v, and your middle finger along w, then stick out your thumb, it will point in the direction of the new vectors.
```

<Figure
    image="figures/introduction-resulting-vector.png" 
/>

<Figure
    image="figures/introduction-right-hand-rule.png" 
/>

> There are some brute-force computational ways to confirm these facts, but I want to share with you a really elegant line of reasoning. This leverages a bit of background, though, so I'm assuming everyone has read chapter 5 on the determinant, and chapter 7 where I introduce the idea of duality. So, you know, go back and take a look if needed.

Reviewer: Make the chapter references lesson links.

<Figure
    image="figures/under-the-light-of-transformations-duel-vector.png" 
/>

The takeaway is that when you're out in the mathematical wild and you find a linear transformation to the number line, you will be able to match it to some vector, which is called the "dual vector" of the transformation, so that performing that linear transformation is the same as taking the dot product with that vector.

## The idea

The cross product gives us a really slick example of this process in action. It takes some effort, but it's definitely worth it. What I'm going to do is define a certain linear transformation from three dimensions to the number line, and it will be defined in terms of two vectors v and w. Then, when we associate that transformation with its dual vector in 3d space, that dual vector will be the cross product of v and w. Understanding that transformation will make clear the connection between the geometry and the computation of the cross product.

<Figure
    image="figures/the-idea-3d-to-1d.png" 
/>

> So, to back up a bit, remember that in two dimensions, computing the 2d version of the cross product of vectors v and w involves taking the determinant of a matrix whose columns contain the coordinates of those vectors. There's no nonsense with basis vectors stuck in the matrix, just an ordinary determinant returning a number. Geometrically, this gives us the area of the parallelogram spanned out by those two vectors, with the possibility of being negative depending on the orientation of the vectors.

Reviewer: ```suggestion
To back up a bit, remember that in two dimensions, computing the 2d version of the cross product of vectors v and w involves taking the determinant of a matrix whose columns contain the coordinates of those vectors. There's no nonsense with basis vectors stuck in the matrix, just an ordinary determinant returning a number. Geometrically, this gives us the area of the parallelogram spanned out by those two vectors, with the possibility of being negative depending on the orientation of the vectors.
```

<!-- TODO:Potentially split the figure or annotate the above figure for another visual here-->

So here we are, out in the mathematical wild with a linear transformation that outputs numbers. The idea of duality should be perking up in your mind. What we're going to do is find the matrix that describes this transformation, which will be a 1x3 matrix since the transformation goes from three dimensions to one. In other words, we're looking for a 1x3 matrix such that multiplying this matrix by some vector x, y, z gives the same result as plugging in x, y, z to the first column of a 3x3 matrix whose other two columns have the coordinates of v and w, and computing the determinant.

<Figure
    image="figures/the-idea-1-by-3-matrix-encoding-the-3d-to-1d-transformation.png" 
/>

If you think in terms of duality and turn this matrix on its side, we're looking for a special 3d vector, that I'll call p, such that taking the dot product between p and any other vector x, y, z gives the same result as plugging in x, y, z to the first column of a matrix and computing the determinant.

<Figure
    image="figures/the-idea-3d-to-1d-transformatio-as-vector.png" 
/>

> I'll get to the geometry of this in just a moment, but right now let's dig in and think about what this means computationally. Taking the dot product between p and [x, y, z] will give something times x, plus something times y, plus something times z, where those somethings are the coordinates of p. Now, when you compute the determinant on the right, you can organize it to look like some constant times x, plus some constant times y, plus some constant times z, where those "somethings" involve certain combinations of the components of v and w.
> 
> <Figure
>     image="figures/the-idea-3d-to-1d-transformation-as-a-vector-computation.png" 
> />
> 
> So the answers to what those "somethings" are will give the coordinates of the vector p that we're looking for.

Reviewer: ```suggestion
I'll get to the geometry of this in just a moment, but right now let's dig in and think about what this means computationally. Taking the dot product between p and [x, y, z] will give

$$(\text{something}) \cdot x + (\text{something}) \cdot y + (\text{something}) \cdot z$$

Those somethings are the coordinates of p. When you compute the determinant on the right, you can organize it to look like 

$$(v_2 w_3 - v_3 w_2) \cdot x + (v_3 w_1 - v_1 w_3) \cdot y + (v_1 w_2 - v_2 w_1) \cdot z$$

This shows what those "somethings" are, and hence what the coordinates of the vector $\mathbf{p}$ are.

<Figure
    image="figures/the-idea-3d-to-1d-transformation-as-a-vector-computation.png" 
/>

```

## Conclusion

So to sum up what just happened, I defined a linear transformation from 3d space to the number line defined in terms of the vectors v and w.

<Figure
    image="figures/conclusion-3d-to-1d-with-variable-vector.png" 
/>

Then I went through two different ways to think about the dual vector of this transformation, the vector such that applying the transformation is the same as taking a dot product with this vector.

<Figure
    image="figures/conclusion-duel-vector-of-this-transformation.png" 
/>

On the one hand, a computational approach will lead you to the trick of plugging in the symbols i hat, j hat and k hat to the first column of a matrix and computing the determinant.

<Figure
    image="figures/conclusion-computational-approach.png" 
/>

But thinking geometrically, we can deduce that this dual vector must be perpendicular to v and w, with a length equal to the area of the parallelogram spanned out by those two vectors. Also, it's direction will be determined by the right-hand rule.

<Figure
    image="figures/conclusion-geometric-approach.png" 
/>

> Next up is a really important concept in linear algebra: Change of basis.

Reviewer: Make "Change of basis" a lesson link.
